spidermon/contrib/scrapy/monitors.py

```diff
@@ -0,0 +1,146 @@
+from spidermon import monitors, MonitorSuite, Monitor
+from spidermon.exceptions import NotConfigured
+from ..monitors.mixins.spider import SpiderMonitorMixin
+
+SPIDERMON_MIN_ITEMS = 'SPIDERMON_MIN_ITEMS'
+SPIDERMON_MAX_ERRORS = 'SPIDERMON_MAX_ERRORS'
+SPIDERMON_EXPECTED_FINISH_REASONS = 'SPIDERMON_EXPECTED_FINISH_REASONS'
+SPIDERMON_UNWANTED_HTTP_CODES = 'SPIDERMON_UNWANTED_HTTP_CODES'
+
+
+class BaseScrapyMonitor(Monitor, SpiderMonitorMixin):
+    longMessage = False
+
+    def get_settings(self, name, default=None):
+        """Return a settings prioritized by:
+
+            #. ``{name}`` in the ``crawler.settings`` object
+            #. If ``crawler.settings.{name}`` is a dict, try to get the
+               ``{spider.name}`` key of that dictionary, this allows you to
+               have a different configuration per spider in a settings level,
+               like:
+
+               .. code-block:: python
+
+                   SPIDERMON_MAX_ERRORS = {
+                        'my_spider': 100,
+                        'other_spider': 10,
+                   }
+
+        """
+        settings_value = self.crawler.settings.get(name.upper())
+        if type(settings_value) == dict:
```

Comment:
I am not 100% certain, but I believe `type(settings_value) is dict` is a better way to check if this variable is a dict.

Reply:
```
isinstance(...)
    isinstance(object, class-or-type-or-tuple) -> bool
```